File: little-newsboy/frontend/src/types/source.ts
export enum SourceType {
  GITHUB = 'github',
  ARXIV = 'arxiv',
  BLOG = 'blog',
  COMMUNITY = 'community',
  NEWS = 'news'
}

export enum UpdateFrequency {
  REALTIME = 'realtime',
  DAILY = 'daily',
  WEEKLY = 'weekly'
}

export enum Priority {
  HIGH = 'high',
  MEDIUM = 'medium',
  LOW = 'low'
}

export enum Status {
  ACTIVE = 'active',
  PAUSED = 'paused'
}

export interface ISource {
  id: string;
  name: string;
  type: SourceType;
  url: string;
  update_frequency: UpdateFrequency;
  priority: Priority;
  status: Status;
  filters?: string;
  credentials?: string;
  user_id: string;
  created_at: string;
  updated_at: string;
}

export interface ISourceCreate {
  name: string;
  type: SourceType;
  url: string;
  update_frequency: UpdateFrequency;
  priority: Priority;
  status: Status;
  filters?: string;
  credentials?: string;
}

export interface ISourceUpdate {
  name?: string;
  type?: SourceType;
  url?: string;
  update_frequency?: UpdateFrequency;
  priority?: Priority;
  status?: Status;
  filters?: string;
  credentials?: string;
}

export interface ISourcesPage {
  items: ISource[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

export const SourceTypeLabels: Record<SourceType, string> = {
  [SourceType.GITHUB]: 'Github',
  [SourceType.ARXIV]: 'Arxiv',
  [SourceType.BLOG]: 'Blog',
  [SourceType.COMMUNITY]: 'Community',
  [SourceType.NEWS]: 'News'
};

export const UpdateFrequencyLabels: Record<UpdateFrequency, string> = {
  [UpdateFrequency.REALTIME]: '实时',
  [UpdateFrequency.DAILY]: '每天',
  [UpdateFrequency.WEEKLY]: '每周'
};

export const PriorityLabels: Record<Priority, string> = {
  [Priority.HIGH]: '高',
  [Priority.MEDIUM]: '中',
  [Priority.LOW]: '低'
};

export const StatusLabels: Record<Status, string> = {
  [Status.ACTIVE]: '活跃',
  [Status.PAUSED]: '暂停'
};

export const StatusColors: Record<Status, string> = {
  [Status.ACTIVE]: 'success',
  [Status.PAUSED]: 'default'
}; 

File: little-newsboy/backend/app/db/models/source.py
from sqlalchemy import Boolean, Column, String, DateTime, JSON, ForeignKey, Enum
from sqlalchemy.sql import func
import uuid
import enum

from app.db.session import Base

class SourceType(str, enum.Enum):
    GITHUB = "github"
    ARXIV = "arxiv"
    BLOG = "blog"
    COMMUNITY = "community"
    NEWS = "news"

class UpdateFrequency(str, enum.Enum):
    REALTIME = "realtime"
    DAILY = "daily"
    WEEKLY = "weekly"

class Priority(str, enum.Enum):
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"

class Status(str, enum.Enum):
    ACTIVE = "active"
    PAUSED = "paused"

class Source(Base):
    __tablename__ = "sources"

    id = Column(String, primary_key=True, index=True, default=lambda: str(uuid.uuid4()))
    name = Column(String, index=True)
    type = Column(Enum(SourceType), index=True)
    url = Column(String)
    update_frequency = Column(Enum(UpdateFrequency), default=UpdateFrequency.DAILY)
    priority = Column(Enum(Priority), default=Priority.MEDIUM)
    status = Column(Enum(Status), default=Status.ACTIVE)
    filters = Column(JSON, default=dict)
    credentials = Column(JSON, default=dict)
    
    user_id = Column(String, ForeignKey("users.id"))
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now()) 